autokeras/tasks/image.py

```diff
@@ -416,3 +416,121 @@ def fit(
+        metrics: A list of Keras metrics. Defaults to use 'accuracy'.
+        project_name: String. The name of the AutoModel.
+            Defaults to 'image_classifier'.
+        max_trials: Int. The maximum number of different Keras Models to try.
+            The search may finish before reaching the max_trials. Defaults to 100.
+        directory: String. The path to a directory for storing the search outputs.
+            Defaults to None, which would create a folder with the name of the
+            AutoModel in the current directory.
+        objective: String. Name of model metric to minimize
+            or maximize, e.g. 'val_accuracy'. Defaults to 'val_loss'.
+        tuner: String or subclass of AutoTuner. If string, it should be one of
+            'greedy', 'bayesian', 'hyperband' or 'random'. It can also be a subclass
+            of AutoTuner. If left unspecified, it uses a task specific tuner, which
+            first evaluates the most commonly used models for the task before
+            exploring other models.
+        overwrite: Boolean. Defaults to `False`. If `False`, reloads an existing
+            project of the same name if one is found. Otherwise, overwrites the
+            project.
+        seed: Int. Random seed.
+        max_model_size: Int. Maximum number of scalars in the parameters of a
+            model. Models larger than this are rejected.
+        **kwargs: Any arguments supported by AutoModel.
+    """
+
+    def __init__(
+        self,
+        num_classes: Optional[int] = None,
+        multi_label: bool = False,
+        loss: types.LossType = None,
+        metrics: Optional[types.MetricsType] = None,
+        project_name: str = "image_classifier",
+        max_trials: int = 100,
+        directory: Union[str, Path, None] = None,
+        objective: str = "val_loss",
+        tuner: Union[str, Type[tuner.AutoTuner]] = None,
+        overwrite: bool = False,
+        seed: Optional[int] = None,
+        max_model_size: Optional[int] = None,
+        **kwargs
+    ):
+        pass
+
+    def fit(
+        self,
+        x: Optional[types.DatasetType] = None,
+        y: Optional[types.DatasetType] = None,
+        epochs: Optional[int] = None,
+        callbacks: Optional[List[tf.keras.callbacks.Callback]] = None,
+        validation_split: Optional[float] = 0.2,
+        validation_data: Union[
+            tf.data.Dataset, Tuple[types.DatasetType, types.DatasetType], None
+        ] = None,
+        **kwargs
+    ):
+        """Search for the best model and hyperparameters for the AutoModel.
+
+        It will search for the best model based on the performances on
+        validation data.
+
+        # Arguments
+            x: numpy.ndarray or tensorflow.Dataset. Training data x. The shape of
+                the data should be (samples, width, height)
+                or (samples, width, height, channels).
+            y: numpy.ndarray or tensorflow.Dataset. Training data y. They are the
```

Comment:
For np, it should be a list of numpy arrays.
For tf dataset, it should be one tf dataset.